src/reanimated2/Sensor.ts

```diff
@@ -0,0 +1,75 @@
+import NativeReanimatedModule from './NativeReanimated';
+import {
+  SensorType,
+  SensorConfig,
+  SharedValue,
+  Value3D,
+  ValueRotation,
+  ShareableRef,
+} from './commonTypes';
+import { makeMutable } from './mutables';
+
+function initSensorData(
+  sensorType: SensorType
+): SharedValue<Value3D | ValueRotation> {
+  if (sensorType === SensorType.ROTATION) {
+    return makeMutable<Value3D | ValueRotation>({
+      qw: 0,
+      qx: 0,
+      qy: 0,
+      qz: 0,
+      yaw: 0,
+      pitch: 0,
+      roll: 0,
+      interfaceOrientation: 0,
+    });
+  } else {
+    return makeMutable<Value3D | ValueRotation>({
+      x: 0,
+      y: 0,
+      z: 0,
+      interfaceOrientation: 0,
+    });
+  }
+}
+
+export default class Sensor<T> {
+  public listenersNumber = 0;
+  private sensorId = -1;
+  private sensorType: SensorType;
+  private data: SharedValue<Value3D | ValueRotation>;
+  private config: SensorConfig;
+
+  constructor(sensorType: SensorType, config: SensorConfig) {
+    this.sensorType = sensorType;
+    this.config = config;
+    this.data = initSensorData(sensorType);
+  }
+
+  register(
+    eventHandler: ShareableRef<T> | ((data: Value3D | ValueRotation) => void)
+  ) {
+    const config = this.config;
+    const sensorType = this.sensorType;
+    this.sensorId = NativeReanimatedModule.registerSensor(
+      sensorType,
+      config.interval === 'auto' ? -1 : config.interval,
+      config.iosReferenceFrame,
+      eventHandler
+    );
+    return this.sensorId !== -1;
+  }
```

Comment:
This part is problematic for me. Why do you need to register a new listener for every sensor when you can use the same shared value for all? We want to have one shared value and one native listener for every `useAnimatedSensor` with the same type.
Based on my tests, it seems that you never unregister the native listeners. This assertion fails on reload:
```
AnimatedSensorModule::~AnimatedSensorModule() {
  assert(sensorsIds_.empty());
}
```